Source/CarthageKit/Xcode.swift

```diff
@@ -893,15 +904,23 @@ public func buildInDirectory( // swiftlint:disable:this function_body_length
 						)
 						.flatMapError { _ in .empty }
 				}
-
-				return createVersionFile(
-					for: dependency.dependency,
-					version: dependency.version,
-					platforms: options.platforms,
-					buildProducts: urls,
-					rootDirectoryURL: rootDirectoryURL
-					)
-					.flatMapError { _ in .empty }
+				var copyToSharedCache = SignalProducer<(URL), CarthageError>.empty
+				if options.cacheBuilds {
+					let sharedCacheURLForDependency = sharedCacheURLByXcodeVersionFor(dependency: dependency.dependency, pinnedVersion: dependency.version)
+					copyToSharedCache = copyFrameworksAndCreateVersionFile(urls, from: rootDirectoryURL, to: sharedCacheURLForDependency,
+																		   projectName: dependency.dependency.name, pinnedVersion: dependency.version, toolchain: options.toolchain)
```

Comment:
This should ignore errors because:

1. If copying to the shared cache fails, the build should still succeed
2. If copying to the shared cache fails, we'll still want to create the version file

Reply:
so I added a `.flatMapError { _ in .empty }` right after `copyFrameworksAndCreateVersionFile` method call